### ISSUE TITLE
Crash in v2.0.0

### ISSUE THREAD

**Issue by fpwong** [fpwong/BlueprintAssistWiki#11]
User experiencing constant freezing when opening blueprint subgraph. Freezes the whole editor and requires forced restart.

**Comment by fpwong** [comment]
Questions:

1. Can you post the crash log (if there is one)?
2. Did this issue only occur from the latest update (2.0.0)?
3. Does it always crash on a certain subgraph, or is it random? If it does always crash on a certain graph could you post that graph.

**Comment by firzgard** [comment]
Hi, the ue4 forum was down for quite some time, so I missed your message.

1. The bug makes the editor frozen without crashing it, so I didn't take a look at the crash log. I will try to reproduce the bug and send you the log later.

2. This issues actually only happed on UE4.23. I haven't used any other BlueprintAssist version aside from 2.0.0 on UE4.23.

3. The bug happens on multiple different subgraphs, seems pretty random to me. However, once a subgraph would cause a freeze, it would be so constantly, even after restart or reenable the plugin. I will send you the screenshots later.

I will try to reproduce the crash and update the info again.

**Comment by firzgard** [comment]
I captured a video of the bug: https://youtu.be/HUUqv_0_NNM

My guess is that the plugin gets stuck in some loop while refreshing the size cache during subgraph entering-exitting, since force-refreshing the size cache with Ctrl + Shift + R will evade the freeze. It will still freeze again when you open it up after restart though.

I checked the log files. the freeze doesn't produce any useful log whatsoever,

**Comment by fpwong** [comment]
Thanks for the video, I will look into this. I actually hadn't tested collapsed graphs in a while since I don't usually use them.

Also I have a feeling the bug may be related to the node size cache file. Could you try setting the `Save Blueprint Assist Cache to File` setting (in Editor Settings) to false and seeing if the crash still occurs.

**Comment by firzgard** [comment]
Hi. Sorry for the delay.  I tried turning off `Save Blueprint Assist Cache to File` but the freeze still occurs.

**Comment by fpwong** [comment]
I was not able to reproduce the bug on my end, but I did find a logic error related to collapsed graphs which I have fixed. Hopefully the update today fixed your bug.

**Comment by firzgard** [comment]
I tried the new version, the freeze still happens at the same graphs.

**Comment by fpwong** [comment]
I'm truly stumped at what is causing this. I have made a duplicate of your first blueprint from the video (the GameplayAbility) but the bug does not occur.

The last possible aspect I am thinking is possibly clearing your node size cache. You should find the cache files in your UE4 install location 

`C:/Program Files/UE_4.23/Engine/Plugins/Marketplace/BlueprintAssist/NodeSizeCache`

Delete all the files inside that folder and try restarting.

If this does not fix it, this bug may not be fixed in a while as I am unable to recreate it (if you can somehow recreate the bug in a simple project and send it to me that would be great).

**Comment by firzgard** [comment]
I did try to reproduce the bug on a new sample project a few days ago, but couldn't.

The bug seems rather random now, there is even 1 time it freezes on opening a state in animblueprint's state machine, but only that 1 time and I couldn't reproduce it again.

Also, I noticed that not only [ctrl + shift + r], [right click + refresh nodes] also do fix the graph temporarily.

Anyway, I guess the refresh-before-open workaround would do for now.
